crates/biome_js_analyze/src/lint/nursery/use_explicit_type.rs

```diff
@@ -202,22 +228,87 @@ declare_lint_rule! {
     /// }
     /// ```
     ///
+    /// The following pattern is considered correct code because it includes both a return type and an argument type on an function:
+    ///
+    /// ```ts
+    /// export function test(a: number, b: number): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// export default function test(obj: {a: string}): void {
+    ///   return;
+    /// }
+    /// ```
+    ///
+    /// ```ts
+    /// export default (a: SomeType): void => {
+    ///   return;
+    /// }
+    /// ```
+    ///
     pub UseExplicitType {
         version: "1.9.3",
         name: "useExplicitType",
         language: "ts",
         recommended: false,
-        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type")],
+        sources: &[RuleSource::EslintTypeScript("explicit-function-return-type"), RuleSource::EslintTypeScript("explicit-module-boundary-types")],
     }
 }
 
 declare_node_union! {
-    pub AnyJsFunctionWithReturnType = AnyJsFunction | JsMethodClassMember | JsMethodObjectMember | JsGetterClassMember | JsGetterObjectMember
+    pub AnyJsFunctionWithReturnTypeOrJsParameters = AnyJsFunction | JsMethodClassMember | JsMethodObjectMember | JsGetterClassMember | JsGetterObjectMember | JsParameters
```

Comment:
I've just noticed that some types are missing for checking if the return type is explicitly set:

- `TsMethodSignatureTypeMember`
- `TsCallSignatureTypeMember`
- `TsMethodSignatureClassMember`
- `TsSetterSignatureClassMember`
- `TsDeclareFunctionDeclaration`
- `TsDeclareFunctionExportDefaultDeclaration`

I wonder if directly matching against `JsParameters` is a good idea. I see two possible alternative approaches:
1. matching directly against `JsFormalParameter`, `JsRestParameter`, and `TsThisParameter`
2. matching against functions with parameters (We already have most of them, we need to add setter and constructors).

Personally I could choose (2) because this allows us to issue a single diagnostic when a parameter or a return type is absent on a function.
Also, similarly to return type, we should ignore callbacks.

Besides, we should think to a better name for this union because we will have to add properties and top-level variables. Maybe `AnyJsTypeableEntity`? Have you a better suggestion?

Reply:
Thanks for the review, and apologies for the late response.

> I've just noticed that some types are missing for checking if the return type is explicitly set:

I understand. I'll create a separate for this this.

> I wonder if directly matching against JsParameters is a good idea. I see two possible alternative approaches:

I see your point. I'll revisit this after implementing the missing return types.


> Maybe AnyJsTypeableEntity? Do you have a better suggestion?

AnyJsTypeableEntity looks good to me.

For now, I will make this PR draft.